dm/pkg/checker/lightning_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checker
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pingcap/tidb/br/pkg/lightning/restore"
+	"github.com/pingcap/tiflow/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+type mockPrecheckItem struct {
+	err  error
+	pass bool
+	msg  string
+}
+
+func (m mockPrecheckItem) Check(ctx context.Context) (*restore.CheckResult, error) {
```

Comment:
I suppose we should add some integratoin test too? The unit test seems only mock the return result.

Reply:
It's hard to generate such test case in the real cluster, I have a manually test with self-built TiDB in PR description